### PR TITLE
mypostcard blog code

### PR DESCRIPTION
**Pull request by nopcoder**

### CODE REVIEW

File: mypostcard/README [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# mypostcard
```
Comment (itaiad200) [inline mypostcard/README:1]: Maybe add a link to the blog once it's out.

File: mypostcard/README [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+Blog's demo example code that uses "embed" directive.
+The service will accept text from the user and generate a postcard image rendered with the text.
+It will include embed the application's license.
```
Comment (itaiad200) [inline mypostcard/README:5]: Not really clear.
```suggestion
It will use embed for the application's license.
```

File: mypostcard/main.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bytes"
+	"embed"
```
Comment (itaiad200) [inline mypostcard/main.go:5]: I would highlight this import by putting it in its own block and add a comment explaining the importance of this import.

File: mypostcard/main.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/freetype/truetype"
+	"golang.org/x/image/font"
+	"golang.org/x/image/math/fixed"
+)
+
+//go:embed LICENSE
```
Comment (itaiad200) [inline mypostcard/main.go:22]: Maybe put in a single block and document nicely?

File: mypostcard/main.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	http.Handle("/", http.FileServer(http.FS(webUI)))
+	http.HandleFunc("/postcard", postcardHandler)
+	fmt.Println("Listen and serve on", *listenAddr)
+	err := http.ListenAndServe(*listenAddr, nil)
+	if err != nil {
+		fmt.Println("ERROR", err)
+	}
+}
+
+func postcardHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (itaiad200) [inline mypostcard/main.go:55]: Document.

File: mypostcard/main.go [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	text := r.FormValue("postcard")
+	text = strings.ReplaceAll(text, "\r", "")
```
Comment (itaiad200) [inline mypostcard/main.go:57]: nit: extract to a func with proper name.

File: mypostcard/main.go [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	w.Header().Set("Content-Type", "image/png")
+	w.Header().Set("Content-Length", strconv.Itoa(len(b.Bytes())))
+	_, err = w.Write(b.Bytes())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func generatePostcard(lines []string) image.Image {
```
Comment (itaiad200) [inline mypostcard/main.go:74]: Prettify this function, add comments and maybe break it up a little.

**Review by itaiad200 — APPROVED** [review]

File: mypostcard/main.go (hunk 4 above)
Comment (nopcoder) [reply mypostcard/main.go:22]: :) it was at first - but wanted to copy-paste specific lines from the code to the blog

File: mypostcard/main.go (hunk 6 above)
Comment (nopcoder) [reply mypostcard/main.go:63]: I'll keep it as one func - a lot of the code here could be reused or extracted into functions but wanted each part as a function for the blog post.

File: mypostcard/main.go (hunk 7 above)
Comment (nopcoder) [reply mypostcard/main.go:81]: There is no golangci-lint for go1.16 yet, but I used `vec` and format using my editor.
I'm keeping the function as a single block for the blog.
If you think there is something specific inside the function that doesn't clear I'll add more information in the function description or line comment.

**Pull request merged**